.github/workflows/ci.yml

```diff
@@ -2,47 +2,59 @@ name: Django CI
-    - name: Install Dependencies
-      run: |
-        docker exec -e TOXENV=${{ matrix.python-version }}-${{ matrix.django-version }} --env DB_HOST=${{ matrix.db-version }} -u root edx_notes_api \
-        /bin/bash -c "apt-get update && apt-get install python3-dev default-libmysqlclient-dev build-essential pkg-config"
-    - name: setup python 311
-      if: ${{ matrix.python-version == 'py311' }}
-      run: |
-        docker exec -e TOXENV=${{ matrix.python-version }}-${{ matrix.django-version }} --env DB_HOST=${{ matrix.db-version }} -u root edx_notes_api \
-        /bin/bash -c "add-apt-repository ppa:deadsnakes/ppa -y && apt install python3.11 python3.11-dev python3.11-distutils -y"    
-    - name: setup python 312
-      if: ${{ matrix.python-version == 'py312' }}
-      run: |
-        docker exec -e TOXENV=${{ matrix.python-version }}-${{ matrix.django-version }} --env DB_HOST=${{ matrix.db-version }} -u root edx_notes_api \
-        /bin/bash -c "add-apt-repository ppa:deadsnakes/ppa -y && apt install python3.12 python3.12-dev python3.12-distutils -y"
-    - name: Run Tests
-      run: |
-        docker exec -e TOXENV=${{ matrix.python-version }}-${{ matrix.django-version }} --env DB_HOST=${{ matrix.db-version }} -u root edx_notes_api /edx/app/edx_notes_api/edx_notes_api/.ci/run_tests.sh
-    - name: Run PII Check
-      run: |
-        docker exec -e TOXENV=${{ matrix.python-version }}-${{ matrix.django-version }} -u root edx_notes_api /edx/app/edx_notes_api/edx_notes_api/.ci/run_pii_checker.sh
-    - name: Run Reserved Keywords Check
-      run: |
-        docker exec -e TOXENV=${{ matrix.python-version }}-${{ matrix.django-version }} -u root edx_notes_api /edx/app/edx_notes_api/edx_notes_api/.ci/run_check_keywords.sh
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install system MySQL
+        run: |
+          sudo apt-get update && sudo apt-get install -y libxmlsec1-dev mysql-client
+          sudo service mysql start
+          mysql -h 127.0.0.1 -u root -proot -e "UPDATE mysql.user SET authentication_string = null WHERE user = 'root'; FLUSH PRIVILEGES;"
+          echo "CREATE DATABASE IF NOT EXISTS edx_notes_api;" | sudo mysql -u root
+
+      - name: Install Elasticsearch
+        run: |
+          curl -fsSL https://artifacts.elastic.co/GPG-KEY-elasticsearch | \
+            sudo gpg --dearmor -o /usr/share/keyrings/elasticsearch-keyring.gpg
+          echo "deb [signed-by=/usr/share/keyrings/elasticsearch-keyring.gpg] \
+            https://artifacts.elastic.co/packages/7.x/apt stable main" | \
+            sudo tee -a /etc/apt/sources.list.d/elastic-7.x.list
+          sudo apt-get update && sudo apt-get install -y elasticsearch=7.13.4
+          sudo service elasticsearch start
```

Comment:
We should use the github actions services rather than doing all this manual installation: https://github.com/openedx/edx-platform/pull/35713